src/Controller/ExportController.php

```diff
@@ -99,8 +95,16 @@ public function home(
 		}
 
 		// If the book title is specified, export it now.
+		$exception = false;
+		$response = new Response();
 		if ( $request->get( 'page' ) ) {
-			return $this->export( $request, $api, $fontProvider, $generatorSelector, $creditRepo );
+			try {
+				return $this->export( $request, $api, $fontProvider, $generatorSelector, $creditRepo );
+			} catch ( WsExportException $ex ) {
+				$exception = $ex;
+				// @TODO Return different status codes for different errors (e.g. timeout vs not found).
+				$response->setStatusCode( 404 );
```

Comment:
If we're going to use a generic response code I would use a 400 or 500. A 404 should strictly be for not found, I think...

If it helps, maybe `WsExportException` could take an optional third argument that sets the status code that should returned here?

Reply:
Yeah I was thinking that would be good; was going to ask about it. Will change it now.